src/server/gdb_server.c

```diff
@@ -166,6 +166,22 @@ struct target *get_available_target_from_connection(struct connection *connectio
 	return target;
 }
 
+/** Return true iff the given connection includes the given target. */
+static bool gdb_connection_includes_target(struct connection *connection, struct target *target)
+{
+	struct gdb_service *gdb_service = connection->service->priv;
+	struct target *service_target = gdb_service->target;
+	if (service_target->smp) {
+		struct target_list *tlist;
+		foreach_smp_target(tlist, service_target->smp_targets) {
+			struct target *t = tlist->target;
+			if (t == target)
+				return true;
+		}
+	}
+	return service_target == target;
```

Comment:
I'm sorry if I'm too pedantic. For SMP targets, the last line `return service_target == target` is redundant. If reached, it will be always `false` for SMP.

To make the situation clearer to the reader, you may consider this structure of code:

```suggestion
	if (service_target->smp) {
		struct target_list *tlist;
		foreach_smp_target(tlist, service_target->smp_targets) {
			struct target *t = tlist->target;
			if (t == target)
				return true;
		}
		return false;
	} else {
		/* Non-SMP target */
		return service_target == target;	
	}
```

Reply:
Done. Thanks.

Reply:
(checkpatch says I'm not allowed to use the `else`)